### PR TITLE
Schema extract scripts PoC

### PR DESCRIPTION
**Pull request by czue**
Probably easiest to review by commit

### CODE REVIEW

File: scripts/schema_extract_case.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    workbook.close()
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print('Usage: python schema_extract_case.py <filename> <sheet name>')
+        sys.exit(1)
+    filename = sys.argv[1]
+    sheet_name = sys.argv[2]
+    run(filename, *extra_args)
```
[the remark this answers is not in the document]
Comment (czue) [reply scripts/schema_extract_case.py]: oops, last minute change I forgot to test, thanks

**Comment by czue** [comment]
@bderenzi can you try again? This is the diff I'm getting now - can fix the `indices` thing if you want - the others I assume is a discrepancy between the `json` you used and the one in DB?

```diff person_case_prod.csv person_case_local.csv 
30a31
> properties.children_died,children_died,,,,,,
43a45
> properties.educational_qualification,educational_qualification,,,,,,
53a56
> properties.is_oos,is_oos,,,,,,
55a59
> properties.last_class_attended_ever,last_class_attended_ever,,,,,,
64a69,70
> properties.num_female_children_died,num_female_children_died,,,,,,
> properties.num_male_children_died,num_male_children_died,,,,,,
83a90
> properties.total_num_children_died,total_num_children_died,,,,,,
85,88c92,96
< properties.indices.household,household,,,,,,
< properties.indices.person,person,,,,,,
< properties.indices.person,person,,,,,,
< properties.indices.ccs_record,ccs_record,,,,,,
---
> properties.was_oos_ever,was_oos_ever,,,,,,
> properties.indices.household,indices.household,,,,,,
> properties.indices.person,indices.person,,,,,,
> properties.indices.person,indices.person,,,,,,
> properties.indices.ccs_record,indices.ccs_record,,,,,,
```

File: scripts/schema_extract_case.py [hunk 2]
```diff
@@ -81,9 +96,10 @@ def _truncate(name, start_char, total_char):
             name = name[:start_char] + '$' + name[-(total_char - (start_char + 1)):]
         return name
 
-    for form in data['group_schemas']:
+    for i, form in enumerate(data['group_schemas']):
         header = main_sheet_name
         parent = _collapse_path(form['path'])
+        default_prefix = PREFIX_MAP[i]
```
Comment (bderenzi) [inline scripts/schema_extract_case.py]: I know very little about the schemas, but just to double check: for cases will there always be the same number of children in `group_schemas` and are they guaranteed to be in that order? Using a static dictionary and relying on the order feels a little fragile if not. 

A quick search through a JSON file example shows that we don't have `properties` further up on the chain. If you feel comfortable that the order will be guaranteed, then seems good.

**Comment by bderenzi** [comment]
This looks good to me! I think leave the `indices.` prefix. It strikes me as more correct to have that in our database.

If I'm reading the diff correctly, the JSON file I gave you in dropbox has **more** fields than the one we used/were using to sync with? 

It strikes me that `educational_qualification` in particular would be quite useful (though probably everything).

**Comment by czue** [comment]
Cool. Confirming that there are in fact more fields in the JSON than the xlsx on the server. And I'm pretty confident about the ordering (and we can obviously fix if there are any issues).

**Pull request merged**